crowdsourcing/viewsets/project.py

```diff
@@ -70,102 +70,192 @@ def list(self, request, *args, **kwargs):
 
         try:
             query = '''
-                    SELECT p.id, p.name, p.description, Max(mod.relevant_requester_rating) FROM (
-
-                    SELECT id, name, description, created_timestamp, last_updated, owner_id, project_id, imputed_rating,
-                        CASE WHEN real_weight IS NULL AND average_requester_rating IS NOT NULL THEN average_requester_rating
-                        WHEN real_weight IS NULL AND average_requester_rating IS NULL THEN 1.99
-                        WHEN real_weight IS NOT NULL AND average_requester_rating IS NULL THEN real_weight
-                        ELSE real_weight + 0.1 * average_requester_rating END relevant_requester_rating
-                        FROM (
-                            SELECT rnk.*, wrr.weight as real_weight, avg.average_requester_rating FROM (
-
-                    --This fetches the modules according to cascading release
-                    SELECT evr.*
-                    FROM(
-                        SELECT avgrat.*, CASE WHEN weight IS NULL
-                            AND adj_average_worker_rating IS NOT NULL THEN adj_average_worker_rating
-                            WHEN weight IS NULL AND adj_average_worker_rating IS NULL THEN 1.99
-                            WHEN weight IS NOT NULL AND adj_average_worker_rating IS NULL THEN weight
-                            ELSE weight + 0.1 * adj_average_worker_rating END worker_relevant_rating
+                    DROP INDEX IF EXISTS orig_targ;
+                    CREATE INDEX orig_targ
+                    ON crowdsourcing_workerrequesterrating (origin_id, target_id);
+
+                    DROP INDEX IF EXISTS wrr_data;
+                    CREATE INDEX wrr_data
+                    ON crowdsourcing_workerrequesterrating (origin_id, target_id, last_updated, origin_type);
+
```

Comment:
database objects are created using migrations not at runtime.. these should be moved to migrations and are executed ONLY once